src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculationInterface, FigureDataInterface {
+    protected String figureName;
+    protected String color;
+
+    public Figure(String color, String figureName) {
+        this.color = color;
+        this.figureName = figureName;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public void draw() {
+        System.out.println("Figure: " + getFigureName() + " area: "
+                + getArea() + "sq. units, side:"
+                + getFigureData() + ", color:"
+                + getColor());
```

Comment:
Avoid using System.out.println in your draw method. It's better to return the drawing information as a String and let the caller decide what to do with it. This will make your code more reusable and easier to test.